udfs/community/csv_to_struct.sql

```diff
@@ -0,0 +1,19 @@
+-- csv_to_struct:
+-- Input:
+-- strList: string that has map in the format a:b,c:d....
+-- Output: struct for the above map.
+CREATE OR REPLACE FUNCTION fn.csv_to_struct(strList STRING)
+AS (
+  CASE
+    WHEN REGEXP_CONTAINS(strList, ',') OR REGEXP_CONTAINS(strList, ':') THEN
```

Comment:
It may be good to have another function which is wrapped by this one such that in the auxiliary function you can pass the csv delimiter and the k/v delimiter. This would make this function more extensible.

Reply:
love it.